nixos/modules/services/networking/networkmanager.nix

```diff
@@ -34,6 +34,8 @@ let
 
     [device]
     wifi.scan-rand-mac-address=${if cfg.wifi.scanRandMacAddress then "yes" else "no"}
+    ${optionalString (cfg.backend == "iwd")
+      ''wifi.backend=iwd''}
```

Comment:
Is the iwd backend enabled at the moment in network manager? https://github.com/NixOS/nixpkgs/issues/45762

Reply:
No, pretty sure the default is wpa_supplicant, tried removing those lines and couldn't reconnect afterwards.